src/components/uploader/directory-tree-line.tsx

```diff
@@ -0,0 +1,75 @@
+import { Button } from 'antd';
+import { DownloadOutlined, DeleteOutlined } from '@ant-design/icons';
+
+import { isMobile } from 'react-device-detect';
+import { useState } from 'react';
+
+type DirectoryTreeLineProps = {
+  disabled: boolean;
+  isLeaf: boolean;
+  name: string;
+  onDownloadClick: () => void;
+  onDeleteClick: () => void;
+};
+
+export const DirectoryTreeLine = ({
+  disabled,
+  isLeaf,
+  name,
+  onDownloadClick,
+  onDeleteClick,
+}: DirectoryTreeLineProps) => {
+  const [hover, setHover] = useState(false);
+  const showButtons = isMobile || hover;
+
+  if (isLeaf) {
+    return (
+      <div
+        style={{
+          display: 'flex',
+          justifyContent: 'space-between',
+          height: '24px',
+        }}
+        onMouseEnter={() => setHover(true)}
+        onMouseLeave={() => setHover(false)}
```

Comment:
Instead of using js for this kind of stuff, what do you think about just add a CSS class and use the `:hover` selector for displaying it?

I am proposing this approach because with JS we can have a situation like this:
![Screenshot 2021-05-04 at 22 50 24](https://user-images.githubusercontent.com/1657091/117068580-c24da700-ad2b-11eb-9907-b37b76c4182f.png)

It happens when you scroll fast the mouse or when the sync is in progress and you do a hover on the line. With the pure CSS approach, we should solve this kind of bug.